java/arcs/core/data/proto/manifest.proto

```diff
@@ -166,6 +166,7 @@ message ListTypeProto {
 // not those resulting from type inference.
 message ConstraintInfo {
   TypeProto constraint_type = 1;
+  bool unconstrained = 2;
```

Comment:
When reading the code, `no_constraints` seems slightly more readable. e.g., `no_constraints` is `true` has some implication on the `constraint_type`.

(Sorry for the back and forth. What do you think?)